clusterloader2/pkg/measurement/common/etcd_metrics.go

```diff
@@ -66,10 +66,14 @@ func (e *etcdMetricsMeasurement) Execute(config *measurement.MeasurementConfig)
 	if err != nil {
 		return nil, err
 	}
-	host, err := util.GetStringOrDefault(config.Params, "host", config.ClusterFramework.GetClusterConfig().GetMasterIp())
+	masterIPs, err := util.GetStringSliceOrDefault(config.Params, "hosts", config.ClusterFramework.GetClusterConfig().GetMasterIps())
 	if err != nil {
 		return nil, err
 	}
+	if masterIPs == nil || len(masterIPs) < 1 {
+		return nil, fmt.Errorf("empty slice: there are no master IPs registered %v", masterIPs)
+	}
+	host := masterIPs[0]
```

Comment:
Actually this is a bit more tricky, If we have more than one masterIp we should monitor each of them, not only the first one. So, we'll need to call e.startCollecting for each masterIp (host) and then rewrite the stopAndSummarize and other parts to make sure we collect and include data from each host in the summary returned by this measurement.